File::Path::rmtree() no longer makes symlink targets world-writable. Patch by Ben Hutchings. (Closes: #487319)

With this, chmod(2) following symlinks is no longer our problem because
$Force_Writeable isn't set on unixish operating systems. This was the
case for the Etch version too.

[rt.cpan.org #36982]
diff --git a/lib/File/Path.pm b/lib/File/Path.pm
index 19b5750..f20424d 100644
--- a/lib/File/Path.pm
+++ b/lib/File/Path.pm
@@ -351,10 +351,8 @@ sub _rmtree {
             }
 
             my $nperm = $perm & 07777 | 0600;
-            if ($nperm != $perm and not chmod $nperm, $root) {
-                if ($Force_Writeable) {
-                    _error($arg, "cannot make file writeable", $canon);
-                }
+            if ($Force_Writeable && $nperm != $perm and not chmod $nperm, $root) {
+                _error($arg, "cannot make file writeable", $canon);
             }
             print "unlink $canon\n" if $arg->{verbose};
             # delete all versions under VMS
